src/main/java/Deadline.java

```diff
@@ -0,0 +1,20 @@
+public class Deadline extends Task {
+    private String byWhen = "";
+
+    public Deadline(String taskName, String byWhen) {
+        super(taskName);
+        this.byWhen = byWhen;
+    }
+
+    public String getByWhen() {
+        return byWhen;
+    }
+
+    public String toString() {
+        if (super.IsCompleted()) {
+            return ".[D][X] " + super.GetTaskName() + "(by: " + getByWhen() + ")";
```

Comment:
You could separate [D] and [X] to different methods, keep [X] in Todo class, then inherit it in Deadline